itinerary_generator/time_utils.py

```
@@ -0,0 +1,163 @@
+    """
+    Convert a datetime to another timezone.
+    
+    Args:
+        dt (datetime): Datetime object to convert (will be made timezone-aware if not already)
+        timezone (str or ZoneInfo): Target timezone
+        
+    Returns:
+        datetime: Datetime in the target timezone
+    """
+    if isinstance(timezone, str):
+        timezone = ZoneInfo(timezone)
```

Ruff (0.8.2)

Local variable `ZoneInfo` referenced before assignment

(F823)